src/unknown_008C40.c

```diff
@@ -162,10 +255,115 @@ void func_800096F8(s32 arg0) {
+    unk80119C50 *temp_v0;
+
+    if (soundMask != 0) {
+        func_800245B4(soundId | 0xE000);
+    }
+    if (D_800DC6E0 == 40) {
+        if (soundMask != 0) {
+            *soundMask = 0;
+        }
+        func_800245B4(-0x55AB);
+        return;
+    }
+    temp_v0 = D_80119C50[D_80119C4C--];
+    temp_v0->x = x;
+    temp_v0->y = y;
+    temp_v0->z = z;
+    temp_v0->soundId = soundId;
+    temp_v0->unk11 = arg4;
+    temp_v0->unk10 = arg5;
+    temp_v0->unkE = arg6;
+    temp_v0->unkF = arg9;
+    temp_v0->unk14 = arg7;
+    temp_v0->unk20 = arg8;
+    temp_v0->unk21 = argA;
+    temp_v0->unk22 = 0;
+    temp_v0->soundMask = soundMask;
+    D_80119C44[D_800DC6E0++] = temp_v0;
+    if (soundMask != 0) {
+        *soundMask = (u32) temp_v0;
+    }
+}
+
+void func_800098A4(u8 arg0, u16 soundId, f32 x, f32 y, f32 z, u8 arg5, u8 arg6,
+                   u8 arg7, u8 arg8, u16 arg9, u8 argA, u8 argB, u8 argC) {
+    Vec3f *temp_a0;
+    unk80119C58 *temp_v1;
+
+    if ((argB < 7) && (argC < 30)) {
+        temp_v1 = &D_80119C58[argB];
+        temp_a0 = &temp_v1->unk4 + argC * 3; //This can't be right...
+        temp_a0->x = x;
+        temp_a0->y = y;
+        temp_a0->z = z;
+        if (argC == 0) {
+            (&D_80119C58[argB])->unk16C = soundId;
+            (&D_80119C58[argB])->unk0.unk0_02 = arg0;
+            (&D_80119C58[argB])->unk170 = arg9;
+            (&D_80119C58[argB])->unk17D = argA;
+            (&D_80119C58[argB])->unk174 = arg6;
+            (&D_80119C58[argB])->unk175 = arg5;
+            (&D_80119C58[argB])->unk176 = arg7;
+            (&D_80119C58[argB])->unk17E = arg8;
+        }
+        if ((&D_80119C58[argB])->unk17C < argC) {
+            (&D_80119C58[argB])->unk17C = argC;
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+// single regalloc diff
+void func_80009968(f32 x, f32 y, f32 z, u8 arg3, u8 arg4, u8 arg5) {
+    Vec3f *temp_a1;
+    unk8011A6D8 *temp_v1;
+
+    if ((arg4 < 7) && (arg5 < 15)) {
+        temp_v1 = &D_8011A6D8[arg4];
+        temp_a1 = (Vec3f *) &temp_v1->unk4 + arg5 * 1;
+        temp_a1->x = x;
+        temp_a1->y = y;
+        temp_a1->z = z;
+        if (arg5 == 0) {
+            temp_v1->unk0.unk0_02 = arg3 & 0xFF;
+        }
+        if (temp_v1->unkB8 < arg5) {
+             temp_v1->unkB8 = arg5;
+        }
+    }
+}
+#else
 GLOBAL_ASM("asm/non_matchings/unknown_005740/func_80009968.s")
-GLOBAL_ASM("asm/non_matchings/unknown_005740/func_800099EC.s")
+#endif
+
+s32 func_800099EC(u8 arg0) {
+    s32 ret;
+    f32 *var_a2;
+    f32 temp_f0;
+    s32 i;
+    unk80119C58 *temp_v0;
+
+    ret = 1;
+    temp_v0 = &D_80119C58[arg0];
+    var_a2 = (f32 *) &temp_v0->unk4;
+    if (temp_v0->unk17C <= 0) {
+        return 0;
+    }
+
+    for (i = 0; i < temp_v0->unk17C; i++) {
+        temp_f0 = *var_a2;
+        if (temp_f0 == -100000.0 || temp_f0 + 1 == -100000.0 || temp_f0 + 2 == -100000.0) {
```

Comment:
Vec3f?

Reply:
Yeah, almost certainly, but no matter what I tried I couldn't get it to match. I likely need to have a better struct definition to make it make sense. This somehow worked though and surprised me as well,